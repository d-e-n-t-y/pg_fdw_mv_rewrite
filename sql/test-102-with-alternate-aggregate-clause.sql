-- Anticipate message "expr (...) not found in MV tlist"
set mv_rewrite.log_match_progress = 'true';

explain (VERBOSE, COSTS OFF) select key, COUNT (key) from test group by key;

select key, COUNT (key) from test group by key;
