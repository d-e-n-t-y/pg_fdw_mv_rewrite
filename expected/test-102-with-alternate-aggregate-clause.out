-- Anticipate message "expr (...) not found in MV tlist"
set mv_rewrite.log_match_progress = 'true';
-- We don't actually care about the plan itself, only the log output.
DO LANGUAGE plpgsql $$BEGIN EXECUTE 'explain (VERBOSE, COSTS OFF) select key, COUNT (key) from test group by key;'; END; $$;
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_mv1
INFO:  mv_rewrite_select_clauses_are_valid_for_mv: expr (count(key)) not found in MV tlist
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_mv2
INFO:  mv_rewrite_join_clauses_are_valid: MV expression ((hidden = 'hidden2'::text)) not found in clauses in the query.
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_expr_mv1
INFO:  mv_rewrite_select_clauses_are_valid_for_mv: expr (count(key)) not found in MV tlist
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_union_mv1
INFO:  mv_rewrite_check_group_clauses_for_mv: looking to rewrite GROUP BY, but MV has no GROUP BY.
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_union_mv2
INFO:  mv_rewrite_check_group_clauses_for_mv: looking to rewrite GROUP BY, but MV has no GROUP BY.
select key, COUNT (key) from test group by key;
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_mv1
INFO:  mv_rewrite_select_clauses_are_valid_for_mv: expr (count(key)) not found in MV tlist
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_mv2
INFO:  mv_rewrite_join_clauses_are_valid: MV expression ((hidden = 'hidden2'::text)) not found in clauses in the query.
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_expr_mv1
INFO:  mv_rewrite_select_clauses_are_valid_for_mv: expr (count(key)) not found in MV tlist
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_union_mv1
INFO:  mv_rewrite_check_group_clauses_for_mv: looking to rewrite GROUP BY, but MV has no GROUP BY.
INFO:  mv_rewrite_add_rewritten_mv_paths: evaluating MV: public.test_union_mv2
INFO:  mv_rewrite_check_group_clauses_for_mv: looking to rewrite GROUP BY, but MV has no GROUP BY.
 key  |  count  
------+---------
 key1 |      10
 key2 | 1000000
(2 rows)

